### PR TITLE
Feature: add migrate command to djangomod

### PR DESCRIPTION
**Pull request by jrbeilke**
### What does this PR do?

Adds support for Django CLI's `migrate` command.

The `syncdb` command was deprecated in Django 1.7 in favor of a new `migrate` command (https://docs.djangoproject.com/en/2.1/releases/1.7/#schema-migrations), and `syncdb` was removed in Django 1.9

### What issues does this PR fix or reference?

#16674

### Tests written?

Yes

### Commits signed with GPG?

No

### CODE REVIEW

**Review by rallytime — CHANGES_REQUESTED** [review]
Welcome @jrbeilke! Thank you for submitting this fix and for writing those tests.

I have one documentation request. Can you add a `.. versionadded:: Neon` tag to the docs of this new function and also document all of the args/kwargs individually? Then we can give this another review.

**Comment by jrbeilke** [comment]
Done and done!

**Review by rallytime — APPROVED** [review]

File: salt/modules/djangomod.py [hunk 1]
```diff
@@ -112,6 +115,70 @@ def syncdb(settings_module,
...
+        Brings the database schema to a state where the named migration is applied,
+        but no later migrations in the same app are applied. This may involve
+        unapplying migrations if you have previously migrated past the named migration.
+        Use the name zero to unapply all migrations for an app.
+
+    database
+        Database to migrate. Defaults to 'default'.
+
+    noinput
+        Suppresses all user prompts. Defaults to True.
```
Comment (gtmanfred) [inline salt/modules/djangomod.py]: can you add docs for all of the options that can be passed in?

**Review by gtmanfred — CHANGES_REQUESTED** [review]
nice, looks great

just one docs nit pick.

File: salt/modules/djangomod.py [hunk 2]
```diff
@@ -112,6 +115,70 @@ def syncdb(settings_module,
...
+
+    migration_name
+        Named migration to be applied to a specific app.
+        Brings the database schema to a state where the named migration is applied,
+        but no later migrations in the same app are applied. This may involve
+        unapplying migrations if you have previously migrated past the named migration.
+        Use the name zero to unapply all migrations for an app.
+
+    database
+        Database to migrate. Defaults to 'default'.
```
Comment (cachedout) [inline salt/modules/djangomod.py]: Should the string `default` be the default argument for the kwarg then?

File: salt/modules/djangomod.py (hunk 1 above)
Comment (jrbeilke) [reply salt/modules/djangomod.py]: Sure, the rest are either default options for django-admin (https://docs.djangoproject.com/en/2.1/ref/django-admin/#default-options), or args that get passed along to cmd.run, but I'll add some quick documentation for them.

File: salt/modules/djangomod.py (hunk 2 above)
Comment (cachedout) [reply salt/modules/djangomod.py]: Gotcha. Thanks for the explanation.

**Review by cachedout — APPROVED** [review]

**Comment by cachedout** [comment]
@gtmanfred Can you re-review this please?

**Review by gtmanfred — APPROVED** [review]

**Pull request merged**